### ISSUE TITLE
xfs still dangerous due to deadlock?

### ISSUE THREAD

**Issue by KlavsKlavsen** [rook/rook#14562]
We are having an issue with hitting inode limits on small PVs that create a lot of small temporary files :(

We wanted to switch to XFS - but then saw this:
https://github.com/rook/rook/blob/fd0c687d6ebb01afc12bc61fc70f03241953ae72/deploy/examples/csi/rbd/storageclass.yaml#L83

Is this still the case.. it would be nice if this linked to the relevant issue - so we could see latest status and possible work arounds (and better understand the issue).. can anyone share any link to an issue explaining this problem? I'll gladly submit PR for improving the example here obviously - to help others who hit this in the future

**Comment by travisn** [comment]
For background on the issue with xfs, see #3132. The ceph tracker linked at the end of that issue does not appear to be resolved.

**Comment by KlavsKlavsen** [comment]
@travisn issue says it should be resolved on kernel side with 5.8 release.. after which a fix in ceph (which I do not know hwat is or if it has been merged yet) should finally resolve it..

**Comment by satoru-takeuchi** [comment]
There was a discussion about the same topic in the rook slack. Raphaël Ducom asked the progress of this issue in ceph-users.

https://lists.ceph.io/hyperkitty/list/ceph-users@ceph.io/thread/XJT4CBZSNFRDA2ZEVJCV4W2NAK3LPZF5/

**Comment by donkeyDau** [comment]
We're also still using ext4 instead of XFS due to this. We're deploying a MongoDB in a hyperconverged scenario (at least  I interpreted this way: the MongoDB pods running on the same node as the OSDs).

If we move the pods to another node: would this enable us to use XFS (as MongoDB recommends)? Or are we still in "danger" to run into this problem then?

**Comment by travisn** [comment]
> We're also still using ext4 instead of XFS due to this. We're deploying a MongoDB in a hyperconverged scenario (at least I interpreted this way: the MongoDB pods running on the same node as the OSDs).
> 
> If we move the pods to another node: would this enable us to use XFS (as MongoDB recommends)? Or are we still in "danger" to run into this problem then?

Correct, until this issue is fixed, if your app pods (mongodb) are running on different nodes from the OSDs, you could safely use XFS.

**Comment by KlavsKlavsen** [comment]
Would it be possible to add some anti-affinity to the chart - so OSDs can't co-exist with PVs using rook-ceph SC ? a workaround granted.. but would ensure its setup correct and making it a default - would mean a value of "allow_xfs_ignoring_bug_14562" or something?

**Comment by travisn** [comment]
Instead of antiaffinity, a common approach is also to label nodes for OSDs, add the [placement.osd.nodeAffinity](https://rook.io/docs/rook/latest-release/CRDs/Cluster/ceph-cluster-crd/#placement-configuration-settings) for the OSDs to run on those nodes, and then apply node affinity or anti-affinity to the application pods accordingly. You could also consider taints on those storage nodes and add `placement.all.tolerations` so all the Rook pods will tolerate it, but not your application pods. All of these custom placement settings can be applied in the helm chart under the `cephClusterSpec`.
